server/src/main/java/org/opensearch/cluster/coordination/RemoteStoreJoinTaskExecutor.java

```diff
@@ -0,0 +1,55 @@
+package org.opensearch.cluster.coordination;
+
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.routing.RerouteService;
+import org.opensearch.cluster.routing.allocation.AllocationService;
+import org.opensearch.common.settings.Settings;
+
+import java.util.List;
+
+import static org.opensearch.action.admin.cluster.remotestore.repository.RemoteStoreRepositoryRegistrationHelper.validateOrAddRemoteStoreRepository;
+import static org.opensearch.action.admin.cluster.remotestore.repository.RemoteStoreRepositoryRegistrationHelper.isRemoteStoreEnabled;
+
+/**
+ * Main executor for Remote Store Nodes joining the OpenSearch cluster
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreJoinTaskExecutor extends JoinTaskExecutor {
```

Comment:
Since we are always using `RemoteStoreJoinTaskExecutor` instead of `JoinTaskExecutor` , does it make sense to move `RemoteStoreJoinTaskExecutor#execute` inside `JoinTaskExecutor` itself ?